packages/text-field/src/TextField/TextField.tsx

```diff
@@ -0,0 +1,135 @@
+import { useComposedRefs } from "@telegraph/compose-refs";
+import { Icon } from "@telegraph/icon";
+import { clsx } from "clsx";
+import React from "react";
+
+import { COLOR, SIZE } from "./TextField.constants";
+
+type TextFieldProps = Omit<
+  React.HTMLAttributes<HTMLInputElement>,
+  "value" | "onChange"
+> & {
+  size?: "1" | "2" | "3";
+  variant?: "outline" | "ghost";
+  disabled?: boolean;
+  error?: boolean | string;
+  value?: string;
+  onChange?: (
+    value: string,
+    { event }: { event: React.ChangeEvent<HTMLInputElement> },
+  ) => void;
+  leadingIcon?: React.ComponentProps<typeof Icon>;
+  TrailingAction?: React.ReactElement;
```

Comment:
Following the direction of the design system here. We could make them take a react element for both but it would be an early abstraction for functionality that's not needed. 

The correct name should be `TrailingAction` as it can accept more than a button, a tooltip for example.

Reply:
> The correct name should be TrailingAction as it can accept more than a button, a tooltip for example.

Ah yep got it.

> We could make them take a react element for both but it would be an early abstraction for functionality that's not needed.

My 2c is the current prop definitions for these two feel perhaps overly specific. Personally it'd be nice to simplify/unify as a start. But not a blocker if you feel strongly about it!